scripts/chat_ui.py

```diff
@@ -0,0 +1,220 @@
+# /// script
+# requires-python = ">=3.12"
+# dependencies = [
+#     "gradio",
+#     "openai",
+# ]
+# ///
+import argparse
+import re
+
+import gradio as gr
+from openai import OpenAI
+
+
+def chat_function(message, history, endpoint, model_name, key, temperature, top_p, max_tokens):
+    """
+    Chat function that communicates with OpenAI-compatible endpoint.
+    """
+    client = OpenAI(api_key=key, base_url=f"http://{endpoint}/v1" if not endpoint.startswith("http") else endpoint)
```

Comment:
does it always recreate the open ai worker when the function is called by gradio ? could makes sense to initiate the client once IMO if possible

Reply:
oh I see good point, lets keep it like this then